app/services/storage_root_migration_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Migrates Complete Moab records to a new Moab Storate Root.
+class StorageRootMigrationService
+  def initialize(from_name, to_name)
+    @from_name = from_name
+    @to_name = to_name
+  end
+
+  # @return [Array<String>] druids of migrated moabs
+  def migrate
+    druids = []
+    from_root.complete_moabs.find_each do |complete_moab|
+      migrate_moab(complete_moab)
+      druids << complete_moab.preserved_object.druid
+    end
+    druids
+  end
+
+  private
+
+  def from_root
+    @from_root ||= MoabStorageRoot.find_by(name: @from_name)
+  end
+
+  def to_root
+    @to_root ||= MoabStorageRoot.find_by(name: @to_name)
+  end
+
+  def migrate_moab(moab)
+    moab.moab_storage_root = to_root
+    moab.status = 'validity_unknown' # This will queue a CV.
+    # Fate of this to be determined by https://github.com/sul-dlss/preservation_catalog/issues/1329
+    moab.last_moab_validation = nil
+    moab.last_checksum_validation = nil
+    moab.last_archive_audit = nil
```

Comment:
one more thought that made it into my branch last night, but not the review i left here:  i'd be in favor of not touching `last_archive_audit`, though i don't feel strongly about it, and could be convinced otherwise.

my reasoning is that the bits that are archived on the cloud are not moving as part of this local moab move, anymore than they are being moved around opaquely under the normal course of our cloud providers' operations.  additionally:
1. the status change doesn't actually trigger `audit_moab_version_replication!` at present anyway, so we'd have to add that hook if that's the desired behavior (not difficult, but a thing we'd need to wire up if we wanted it).
2. the cloud archive checks don't pull anything down for fixity checking or comparison to our local moab copies.  so even if the status change did trigger an audit, i'm not sure the moab move itself would have any bearing, other than that we'd be taking an opportunity to audit more than we otherwise would.

so i think i am slightly opposed to this for both semantic and pragmatic reasons, but like i said, not strongly, and i'm very open to counter-argument if i'm missing something.